Automattic/jetpack
# Extract and move the UpsellBanner component to js-packages/components (2/3)

## Proposed changes:

This PR extracts the UpsellBanner from `packages/my-jetpack` into `js-packages/components`.

- Move the component to `js-packages/components`.
- Rewrite the component in TypeScript.
- Add the UpsellBanner Storybook.
- Update the JetpackManageBanner to use the UpsellBanner from `@automattic/jetpack-components` and move it into the parent folder.

![image](https://github.com/Automattic/jetpack/assets/9832440/8d01d876-fd90-49eb-87af-b0bdff26c04e)

### Other information:


## Jetpack product discussion


Previous PR: https://github.com/Automattic/jetpack/pull/35223

## Does this pull request change what data or activity we track or use?



## Testing instructions:
- Prepare your environment, build: `jetpack build packages/my-jetpack` and `jetpack build js-packages/components`
- Check the code
- Test it. You have to see the banner in My Jetpack (you should have your testing site connected to WPCOM and have 2 or more sites connected to WPCOM.
